src/librustc/traits/trans/mod.rs

```diff
@@ -0,0 +1,73 @@
+use dep_graph::{DepGraph, DepNode, DepTrackingMap, DepTrackingMapConfig};
+use hir::def_id::DefId;
+use std::cell::RefCell;
+use std::marker::PhantomData;
+use traits::Vtable;
+use ty::{self, Ty};
+
+/// Specializes caches used in trans -- in particular, they assume all
+/// types are fully monomorphized and that free regions can be erased.
```

Comment:
> I more or less want to kill all "trans" terminology and just have backends that use Reveal::All and do the lifetime erasing themselves, in the long term.

I basically agree, but I'm not sure I want to do that now. In particular, I want to remove these caches eventually and just sort of roll this into a more general caching scheme. Therefore, I tried to just shuffle things around for now rather than worry too much about the details, so that we can address them in a more uniform fashion.